From 4b4770236ca46dea95475568ea570bbcf0d7edca Mon Sep 17 00:00:00 2001
From: jiajia Qian <jiajia.qian@nxp.com>
Date: Sat, 8 Nov 2025 17:23:19 +0800
Subject: [PATCH] RVGPU/FW: prioritize softpipe over llvmpipe for swrast

This patch modifies the swrast driver dependency to exclude llvmpipe

when both softpipe and llvmpipe are enabled. The goal is to ensure

softpipe is used as the default software rasterizer, avoiding automatic

fallback to llvmpipe during runtime.

Upstream-Status: Inappropriate [i.MX specific]
Signed-off-by: jiajia Qian <jiajia.qian@nxp.com>
---
 src/gallium/meson.build | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/gallium/meson.build b/src/gallium/meson.build
index c26e98e6f54..e31571c47c1 100644
--- a/src/gallium/meson.build
+++ b/src/gallium/meson.build
@@ -61,7 +61,7 @@ else
 endif
 if with_any_llvmpipe and with_gallium_softpipe
   driver_swrast = declare_dependency(
-    dependencies : [ driver_softpipe, driver_llvmpipe ],
+    dependencies : [ driver_softpipe],
   )
 elif with_any_llvmpipe
   driver_swrast = driver_llvmpipe
-- 
2.34.1

